ir/inst_aggregate.go

```diff
@@ -92,8 +92,13 @@ type InstInsertValue struct {
 // NewInsertValue returns a new insertvalue instruction based on the given
 // aggregate value, element and indicies.
 func NewInsertValue(x, elem value.Value, indices ...uint64) *InstInsertValue {
+	elemType := aggregateElemType(x.Type(), indices)
+	if !elemType.Equal(elem.Type()) {
+		panic(fmt.Errorf("insertvalue elem type mismatch, got %v, want %v", elem.Type(), elemType))
```

Comment:
For consistency with other panics of the same nature, we may want to standardize on a format for these error messages. I'm fine with either, so long as we are consistent.

Previous example uses `expected Foo, got Bar`:

> constant/expr_other.go:			panic(fmt.Errorf("invalid icmp operand type; expected *types.IntType, *types.PointerType or *types.VectorType, got %T", xType))

Reply:
Moved to #70. Please continue to review closely and point these out, I need help getting the error messages right. I have been aware of this but evidently didn't succeed here.